### PR TITLE
init refactor testapp

### PR DESCRIPTION
**Pull request by satawatnack**
## Implementation details
- refactor tesapp with more structure and change file name to the standard cosmos chain `test_helpers.go`

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: testing/testdata/wasm_util.go [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 
-func compile(code []byte) []byte {
-	compiled, err := OwasmVM.Compile(code, types.MaxCompiledWasmCodeSize)
+func Compile(code []byte) []byte {
+	owasmVM, err := owasm.NewVm(10)
+	if err != nil {
+		panic(err)
+	}
+
+	compiled, err := owasmVM.Compile(code, types.MaxCompiledWasmCodeSize)
```
Comment (colmazia) [inline testing/testdata/wasm_util.go:18]: is the Owasm in test_helpers is still in use

File: testing/test_helpers.go [hunk 2]
```diff
@@ -137,8 +120,25 @@ func (app *TestingApp) GetTxConfig() client.TxConfig {
 	return bandapp.MakeEncodingConfig().TxConfig
 }
 
+// DefaultConsensusParams defines the default Tendermint consensus params used in TestingApp.
+var DefaultConsensusParams = &tmproto.ConsensusParams{
```
Comment (colmazia) [inline testing/test_helpers.go:124]: why move?

File: benchmark/helper_test.go [hunk 3]
```diff
@@ -9,7 +9,8 @@ import (
 	"time"
 
 	"github.com/bandprotocol/chain/v2/pkg/obi"
-	"github.com/bandprotocol/chain/v2/testing/testapp"
+
+	bandtesting "github.com/bandprotocol/chain/v2/testing"
 	oracletypes "github.com/bandprotocol/chain/v2/x/oracle/types"
```
Comment (RogerKSI) [inline benchmark/helper_test.go:14]: move our package to bottom of import.

File: x/oracle/keeper/data_source_test.go [hunk 4]
```diff
@@ -134,33 +148,37 @@ func TestEditDataSourceDoNotModify(t *testing.T) {
...
 
 func TestAddDataSourceDataSourceMustReturnCorrectID(t *testing.T) {
-	_, ctx, k := testapp.CreateTestInput(true)
+	app, ctx := bandtesting.CreateTestApp(t, true)
+	k := app.OracleKeeper
+
 	// Initially we expect the data source count to be what we have on genesis state.
-	genesisCount := uint64(len(testapp.DataSources)) - 1
+	genesisCount := uint64(len(bandtesting.DataSources)) - 1
+	fmt.Println(genesisCount, k.GetDataSourceCount(ctx))
```
Comment (RogerKSI) [inline x/oracle/keeper/data_source_test.go:156]: why print it here?

**Comment by RogerKSI** [comment]
please help check conflict.

File: testing/testdata/wasm_util.go (hunk 1 above)
Comment (satawatnack) [reply testing/testdata/wasm_util.go:18]: used in line 549

**Comment by RogerKSI** [comment]
@satawatnack Can add more detail on PR? like what changed? idea?

**Review by RogerKSI — APPROVED** [review]

**Review by RogerKSI — APPROVED** [review]

**Review by taobun — APPROVED** [review]

File: testing/test_helpers.go [hunk 5]
```diff
@@ -422,33 +264,34 @@ func setup(withGenesis bool, invCheckPeriod uint, chainID string) (*TestingApp,
...
+		100,
+		baseapp.SetSnapshot(snapshotStore, snapshotOptions),
+		baseapp.SetChainID(chainID),
+	)
 	if withGenesis {
 		return app, bandapp.NewDefaultGenesisState(), dir
 	}
-	return app, bandapp.GenesisState{}, dir
+
+	return app, bandapp.GenesisState{}, ""
```
Comment (colmazia) [inline testing/test_helpers.go:282]: should return dir naa

Comment (RogerKSI) [reply testing/test_helpers.go]: fixed

**Review by colmazia — APPROVED** [review]

**Pull request merged**